akka-stream/src/main/scala/akka/stream/impl/fusing/AggregateWithTimeBoundary.scala

```diff
@@ -72,28 +64,24 @@ class AggregateWithBoundary[T, Agg, Emit](
         val input = grab(in)
         if (aggregated == null) aggregated = allocate
         if (aggregate(aggregated, input)) harvestAndEmitOrEnqueue()
-        if (!bufferFull) pull(in)
+        if (!backPressure) pull(in)
       }
 
       override def onUpstreamFinish(): Unit = {
-        if (buffer.nonEmpty) emitMultiple(out, buffer.iterator)
         if (aggregated != null) emit(out, harvest(aggregated))
```

Comment:
Since I couldn't remember, perhaps this one is tricky for others as well, wdyt about a small reminder here:
```suggestion
        // Note that emit will keep the stage alive until event is emitted 
        if (aggregated != null) emit(out, harvest(aggregated))
```

Reply:
basically the fact that `emit` is asynchronous, unlike `push` right?

Reply:
yes, async in some sense and stateful in that it actually switches the output handler until done